subsys/net/lib/coap/coap_client.c

```diff
@@ -591,6 +615,18 @@ struct coap_client_internal_request *get_request_with_token(struct coap_client *
 	return NULL;
 }
 
+static bool find_echo_option(const struct coap_packet *response, struct coap_option *option)
+{
+	int count = 1;
+
+	count = coap_find_options(response, COAP_OPTION_ECHO, option, count);
```

Comment:
I'd consider simplifying the overall function body, it could easily be a single-liner w/o loss in readability IMO (`return coap_find_options() > 0;`)